scalableminds/webknossos
# Add Recommended Configuration to Task Types

This PR allows to add a recommended configuration to task types. The recommended config is shown to users when they start to trace a task and their configuration differs from the recommended one. Users can then accept or decline this config which automatically applies it (or not).

The message will not trigger twice if the "Finish and get next task" functionality is used and the user changes settings in the mean time. It will also not trigger if a tracing is continued. However, if the user returns to the dashboard and starts the next task from there, the message will trigger again, although the task type did not change.
The requirement (from discuss) was:

> These settings would then be set whenever a hiwi get a new task type.
again, only once when the task type changes.

but I'm not sure how to achieve that. We would need to persist the last task type for that.
With the current implementation, I'm afraid users might get annoyed with the configuration recommendations. What do you think?

### URL of deployed dev instance (used for testing):
- https://recommendedsettings.webknossos.xyz

### Steps to test:
- Create a new task type. Select to "Add Recommended User Settings" and play around with the values, json in the textarea. You should not be able to create the task type with an invalid configuration.
- Once you created a task type with a recommended config, click "Edit" in the task type list. The recommended config field should be checked and expanded.
- During task type creation, if the recommended config contains errors and you uncheck that you want to add a recommended config, you should still be able to create the task type.

- Create a task with a task type with/without a recommended config and get this task as a user.
- Open the task to trace. You should see a message informing you about the recommended settings that can be applied or not.
- Once you started tracing and reload the page, the message should no longer appear.

### Issues:
- fixes #3363 

------
